### PR TITLE
Implement inscureEdgeTermination options for reencrypt and pasthrough routes

### PR DESCRIPTION
**Pull request by JacobTanenbaum**
reencrypt routes work the same as edge routes with Allow, Redirect, and
None

passthrough routes are only allowed Redirect or None

change the name of the haproxy acl's and map files to makes sense

map files
os_edge_http_redirect -> os_route_http_redirect
os_edge_http_expose   -> os_route_http_expose

acls

edge_http_expose          -> route_http_expose
wildcard_edge_http_expose -> wildcard_route_http_expose

git issue: [11547](https://github.com/openshift/origin/issues/11547)
bug: 1317159 [Link](https://bugzilla.redhat.com/show_bug.cgi?id=1317159)
docs: [PR3244](https://github.com/openshift/openshift-docs/pull/3244)

### CODE REVIEW

**Comment by JacobTanenbaum** [comment]
@knobunc

**Comment by knobunc** [comment]
@openshift/networking PTAL

**Comment by knobunc** [comment]
Also, I presume there are docs that need to be updated too?

File: pkg/route/api/validation/validation.go [hunk 1]
```diff
@@ -264,23 +264,24 @@ func validateInsecureEdgeTerminationPolicy(tls *routeapi.TLSConfig, fldPath *fie
...
 	allowedValues := map[routeapi.InsecureEdgeTerminationPolicyType]struct{}{
 		routeapi.InsecureEdgeTerminationPolicyNone:     {},
 		routeapi.InsecureEdgeTerminationPolicyAllow:    {},
 		routeapi.InsecureEdgeTerminationPolicyRedirect: {},
 	}
 
-	if _, ok := allowedValues[tls.InsecureEdgeTerminationPolicy]; !ok {
-		msg := fmt.Sprintf("invalid value for InsecureEdgeTerminationPolicy option, acceptable values are %s, %s, %s, or empty", routeapi.InsecureEdgeTerminationPolicyNone, routeapi.InsecureEdgeTerminationPolicyAllow, routeapi.InsecureEdgeTerminationPolicyRedirect)
-		return field.Invalid(fldPath, tls.InsecureEdgeTerminationPolicy, msg)
+	if routeapi.TLSTerminationEdge == tls.Termination || routeapi.TLSTerminationReencrypt == tls.Termination {
```
Comment (smarterclayton) [inline pkg/route/api/validation/validation.go]: Style: Make this a switch statement on the termination field

File: pkg/route/api/validation/validation.go [hunk 2]
```diff
@@ -264,23 +264,27 @@ func validateInsecureEdgeTerminationPolicy(tls *routeapi.TLSConfig, fldPath *fie
...
 	allowedValues := map[routeapi.InsecureEdgeTerminationPolicyType]struct{}{
 		routeapi.InsecureEdgeTerminationPolicyNone:     {},
 		routeapi.InsecureEdgeTerminationPolicyAllow:    {},
 		routeapi.InsecureEdgeTerminationPolicyRedirect: {},
 	}
 
-	if _, ok := allowedValues[tls.InsecureEdgeTerminationPolicy]; !ok {
-		msg := fmt.Sprintf("invalid value for InsecureEdgeTerminationPolicy option, acceptable values are %s, %s, %s, or empty", routeapi.InsecureEdgeTerminationPolicyNone, routeapi.InsecureEdgeTerminationPolicyAllow, routeapi.InsecureEdgeTerminationPolicyRedirect)
-		return field.Invalid(fldPath, tls.InsecureEdgeTerminationPolicy, msg)
+	switch tls.Termination {
```
Comment (JacobTanenbaum) [inline pkg/route/api/validation/validation.go]: @marun updated the unit test

**Comment by knobunc** [comment]
And are there docs we need to update?

**Comment by ramr** [comment]
LGTM

**Comment by knobunc** [comment]
@marun are you good with the test case he added?

**Comment by JacobTanenbaum** [comment]
@marun did I implement the kind of testing you where looking for?

**Comment by marun** [comment]
LGTM [test]

**Comment by knobunc** [comment]
[merge]

**Comment by knobunc** [comment]
[test] epel yum install failed

**Comment by knobunc** [comment]
[merge] Last flaked on https://github.com/openshift/origin/issues/11004

**Pull request merged**